core/src/main/scala/kafka/log/ProducerStateManager.scala

```diff
@@ -753,6 +753,18 @@ class ProducerStateManager(val topicPartition: TopicPartition,
       None
   }
 
-  private def listSnapshotFiles: Seq[File] = ProducerStateManager.listSnapshotFiles(logDir)
+  private[log] def listSnapshotFiles: Seq[File] = ProducerStateManager.listSnapshotFiles(logDir)
 
+  /**
+   * Remove any producer state snapshot files which do not have a corresponding offset provided
+   * in keepOffsets. The latest snapshot file will always be kept.
+   */
+  def cleanupOrphanSnapshotFiles(segmentBaseOffsets: Iterable[Long]): Unit = {
```

Comment:
Thanks @junrao,

I wanted to enumerate why it's safe to remove these now to make sure we're on the same page.

- `deleteSnapshotFiles` can be deleted because snapshot file deletion is done outside of the ProducerStateManager now.

- `truncate` can be deleted because we only ever fully truncate the producer state snapshots if we are also truncating the log, and as a result, the segment files.

- `truncateAndReload` and `truncateHead` seems to be called when recovering a segment and when loading the log. Upon starting up after an unclean shutdown, we will call `truncateAndReload`. I'm a bit less certain that we can get rid of these methods, I will need to investigate further.